### DISCUSSION TITLE
output Hirshfeld charges

### DISCUSSION THREAD

**Discussion by Satinelamp** [deepmodeling/abacus-develop#5286] [General]
### Background

Hirshfeld charges is useful to analysis chemical reaction.

### Describe the solution you'd like

output Hirshfeld charges during MD at each step.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by QuantumMisaka** [comment]
I do consider that ABACUS need more population analysis method out of Mulliken

**Comment by Satinelamp** [comment]
Here are more options for charge analysis.
https://www.whxb.pku.edu.cn/CN/abstract/abstract27818.shtml
